datafusion/sql/src/unparser/plan.rs

```diff
@@ -587,12 +587,24 @@ impl Unparser<'_> {
                     builder = builder.project(project_columns)?;
                 }
 
-                let filter_expr = table_scan
+                let filter_expr: Result<Option<Expr>> = table_scan
                     .filters
                     .iter()
                     .cloned()
-                    .reduce(|acc, expr| acc.and(expr));
-                if let Some(filter) = filter_expr {
+                    .map(|expr| {
+                        if let Some(ref mut rewriter) = filter_alias_rewriter {
+                            expr.rewrite(rewriter).data()
+                        } else {
+                            Ok(expr)
+                        }
+                    })
+                    .reduce(|acc, expr_result| match (acc, expr_result) {
+                        (Ok(acc_expr), Ok(expr)) => Ok(acc_expr.and(expr)),
+                        (Err(e), _) | (_, Err(e)) => Err(e),
+                    })
```

Comment:
We can simplify here like this
```suggestion
                    .reduce(|acc, expr_result| {
                        acc.and_then(|acc_expr| {
                            expr_result.map(|expr| acc_expr.and(expr))
                        })
                    })
```